addons/stock/models/stock_picking.py

```diff
@@ -604,6 +610,30 @@ def _compute_show_validate(self):
             else:
                 picking.show_validate = True
 
+    @api.depends('state', 'move_lines')
+    def _compute_show_allocation(self):
+        self.show_allocation = False
+        if not self.user_has_groups('stock.group_reception_report'):
+            return
+        for picking in self:
+            if not picking.picking_type_id or picking.picking_type_id.code == 'outgoing':
+                continue
+            else:
+                lines = picking.move_lines.filtered(lambda m: m.product_id.type == 'product' and m.state != 'cancel')
+                if lines:
+                    allowed_states = ['confirmed', 'partially_available', 'waiting']
+                    if picking.state == 'done':
+                        allowed_states += ['assigned']
+                    wh_location_ids = self.env['stock.location'].search([('id', 'child_of', picking.picking_type_id.warehouse_id.view_location_id.id), ('location_id.usage', '!=', 'supplier')]).ids
```

Comment:
`('location_id.usage', '!=', 'supplier')`, it means : Don't take location where the parent location is a supplier.
It shouldn't be `('usage', '!=', 'supplier')` ?